torch/distributed/distributed_c10d.py

```
@@ -2797,3 +2791,124 @@ def new_group(ranks=None, timeout=default_pg_timeout, backend=None, pg_options=N
+            will provide errors to the user which can be caught and handled,
+            but due to its blocking nature, it has a performance overhead. On
+            the other hand, ``NCCL_ASYNC_ERROR_HANDLING`` has very little
+            performance overhead, but crashes the process on errors. This is
+            done since CUDA execution is async and it is no longer safe to
+            continue executing user code since failed async NCCL operations
+            might result in subsequent CUDA operations running on corrupted
+            data. Only one of these two environment variables should be set.
+        backend (str or Backend, optional): The backend to use. Depending on
+            build-time configurations, valid values are ``gloo`` and ``nccl``.
+            By default uses the same backend as the global group. This field
+            should be given as a lowercase string (e.g., ``"gloo"``), which can
+            also be accessed via :class:`Backend` attributes (e.g.,
+            ``Backend.GLOO``). If ``None`` is passed in, the backend
+            corresponding to the default process group will be used. Default is
+            ``None``.
+        pg_options (ProcessGroupOptions, optional): process group options
+            specifying what additional options need to be passed in during
+            the construction of specific process groups. i.e. for the ``nccl``
+            backend, is_high_priority_stream can be specified so that process
+            group can pick up high priority cuda streams.
+
+    Returns:
+        The subgroup containing the current rank, and all the subgroups used for cleanup.
+    """
+    subgroups = []
+    cur_subgroup = None
+    if ranks_per_subgroup_list:
+        for ranks in ranks_per_subgroup_list:
+            subgroup = new_group(
+                ranks=ranks, timeout=timeout, backend=backend, pg_options=pg_options
+            )
+            subgroups.append(subgroup)
+
+            rank = get_rank()
```

also, what happens if a rank given in `ranks` is not part of the given `group`? Should we throw an appropriate error? Can we also add a test for this condition?

---

Added a check and a test `test_new_subgroups_input_rank_exceeds_world_size_nccl`.